routers/web/repo/editor.go

```diff
@@ -151,22 +150,22 @@ func editFile(ctx *context.Context, isNewFile bool) {
 			return
 		}
 
-		dataRc, err := blob.DataAsync()
+		buf, dataRc, fInfo, err := getFileReader(ctx, ctx.Repo.Repository.ID, blob)
 		if err != nil {
-			ctx.NotFound(err)
+			if git.IsErrNotExist(err) {
+				ctx.NotFound(err)
+			} else {
+				ctx.ServerError("getFileReader", err)
+			}
 			return
 		}
 
 		defer dataRc.Close()
 
 		ctx.Data["FileSize"] = blob.Size()
 
-		buf := make([]byte, 1024)
-		n, _ := util.ReadAtMost(dataRc, buf)
-		buf = buf[:n]
-
 		// Only some file types are editable online as text.
-		if !typesniffer.DetectContentType(buf).IsRepresentableAsText() {
+		if !fInfo.isTextFile || fInfo.isLFSFile {
```

Comment:
`IsRepresentableAsText` includes more types like `IsSvgImage`

But `Info.isTextFile` only detects `text/xxx`?

Reply:
You're right, that's a regression

Reply:
#34374

Reply:
Thank you very much!